main.js

```diff
@@ -121,8 +121,7 @@ app = {
 				// check to see if it's new
 				if (hackProgress === newHackProgress) {
 					// the bar hasn't moved
-					log("* Progress bar hasn't moved, waiting");
-					// TODO: make this an automatic process
+					app.restart();	//restart if not work
```

Comment:
Although I *do* want it to be an automatic process, I want more intelligence than just restarting.  I tried just having it restart, but wasn't pleased with the results.